resources/indexer.py

```diff
@@ -480,30 +511,45 @@ def _prep_l2_dir_metadata(self, path):
 
     def new_file(self, file):
         """Factory method for returning different metadata-file types"""
```

Comment:
Preface: No need to address this comment for this review. This is one those highfalutin comments that suggests vast changes while assuming all of that effort doesn't require any effort.

I'm going to sketch out (this is the hand-wavy part where the effort gets hidden) a different way of doing this:
- Create a factory registry list like `METADATA_TYPE_FACTORY = []` near the top.
- Convert the `is_file()` static methods of each class into factory methods that return either an object of the class, or `None`, based on the file path provided.
- Have each class register its factory method with the registry list. (Here's the really bogus effort hide: How do we ensure the factory for BasicFileMetadata ends up at the end of the registry list?)
- Turn this `new_file` function into a simple loop over the registered factory methods;
```for factory in METADATA_TYPE_FACTORY:
    my_file = factory()
    if my_file:
        return my_file
    return None  # or raise Exception
```

Again, no need to actually address this comment. It's more just "here's another possibility to consider".

Reply:
I actually really like this idea. I avoided this pattern because of the extra work I do in check L2 if-block.